### PR TITLE
Added fields, updated serializer for personal data on profile

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Refs #149 

#### What's this PR do?
 - Adds some fields for personal data in profile, removes year_of_birth to be replaced by date_of_birth
 - Renames fields we don't plan on using to have `edx_` prefix. This includes certain job or school information which we'll deal with in a later PR
 - Populates some of these new fields with information from edX. first_name and last_name are created by putting the first word in first_name and all other words in last_name
 - Adds tests for dashboard `SETTINGS`
 - Some fields removed from serializer which we don't plan on using

#### Where should the reviewer start?
profiles/models.py

#### How should this be manually tested?
Run database migrations, go to `/api/v0/profiles/<username>/` and see the updated fields. You should also be able to PATCH most fields.

#### Any background context you want to provide?
This is part of #149, the UI portion will be in a separate PR

### CODE REVIEW

File: profiles/util_test.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        assert first_name == "two"
+        assert last_name == "names"
+
+    def test_split_name_more_than_two(self):
+        """
+        Split name should be limited to two names
+        """
+        first_name, last_name = split_name("three names here")
+        assert first_name == "three"
+        assert last_name == "names here"
```
Comment (giocalitri) [inline profiles/util_test.py]: I do not think this is an easy function to implement, but there are people with multiple first name (and no middle).
Not sure we care at all

**Comment by giocalitri** [comment]
this looks pretty good: a single question.
I did not tests the functionality yet

File: profiles/util_test.py (hunk 1 above)
Comment (noisecapella) [inline profiles/util_test.py]: `split_name` is just used to create inital first and last names, they will have the opportunity to adjust this if we get it wrong during the profile edit.

**Comment by giocalitri** [comment]
I am not sure when this problem was introduced, but `account_privacy` in edx can have a value of `all_users` that does not seem to be allowed any more.

**Comment by giocalitri** [comment]
functionality, other than the bug I found with the field coming from edX, is fine :+1:

**Comment by noisecapella** [comment]
There are different settings for account privacy for edX and MM but we forgot to translate the account privacy correctly. I'll fix that and add a test for it

**Pull request merged**